chameleon-core/src/main/java/org/hibernate/omm/ast/mql/MutationMQLTranslator.java

```diff
@@ -282,12 +292,35 @@ protected void renderUpdateClause(final UpdateStatement updateStatement) {
     @Override
     protected void renderSetClause(final List<Assignment> assignments) {
         try {
+            List<AstFieldUpdate> updates = new ArrayList<>();
             getClauseStack().push(Clause.SET);
             for (Assignment assignment : assignments) {
-                visitSetAssignment(assignment);
+                updates.add(mqlAstState.expect(AttachmentKeys.fieldUpdate(), () -> visitSetAssignment(assignment)));
             }
+            mqlAstState.attach(AttachmentKeys.fieldUpdates(), updates);
         } finally {
             getClauseStack().pop();
         }
     }
+
+    @Override
+    protected void visitSetAssignment(final Assignment assignment) {
+        final List<ColumnReference> columnReferences = assignment.getAssignable().getColumnReferences();
+        if (columnReferences.size() == 1) {
+            ColumnReference columnReference = columnReferences.get(0);
+            if (columnReference.getQualifier() != null) {
+                // TODO: anything to do here?
```

Comment:
very likely we need to attend to it in the future, e.g. in case table joining is involved then we might need to pin down which field to be updated. but for now leaving a comment there suffices.